pkg/volume/volume_linux_test.go

```diff
@@ -451,6 +456,72 @@ func TestSetVolumeOwnershipOwner(t *testing.T) {
 	}
 }
 
+func TestSetVolumeOwnershipOwnerLongTime(t *testing.T) {
+	fsGroup := int64(3000)
+
+	// Change podFsGroupCheckInterval to 1s to reduce test duration
+	podFsGroupCheckInterval = 2 * time.Second
+
+	fakePod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "default",
+		},
+	}
+
+	fakeRecorder := record.NewFakeRecorder(100)
+	eventRecorderFunc := func(message string) {
+		fakeRecorder.Eventf(fakePod, v1.EventTypeWarning, kevents.SlowVolumeFSGroup, message, "fake-volume")
+	}
+
+	tests := []struct {
+		description      string
+		fsGroup          *int64
+		expectedEvents   int
+		mockWalkDeepFunc func(root string, walkFunc filepath.WalkFunc) error
+	}{
+		{
+			description:      "FsGroup time is less than podFsGroupCheckInterval",
+			fsGroup:          &fsGroup,
+			expectedEvents:   0,
+			mockWalkDeepFunc: func(root string, walkFunc filepath.WalkFunc) error { return nil },
+		},
+		{
+			description:    "FsGroup time is greater than podFsGroupCheckInterval",
+			fsGroup:        &fsGroup,
+			expectedEvents: 1,
+			mockWalkDeepFunc: func(root string, walkFunc filepath.WalkFunc) error {
+				timeClock.Sleep(3 * time.Second)
```

Comment:
I think you can remove the sleep by using FakeClock and FakeTimer. The [Step() function](https://github.com/kubernetes/utils/blob/3b25d923346b/clock/testing/fake_clock.go#L178) will advance the clock and trigger timers.

Although I think there may still be a race between when `mockWalkDeepFunc` returns and when we check for events